src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -1941,4 +2309,71 @@ class ActionDelegate {
         }
     }
 
+    String makeUrl(String publicViewUrl = publicViewProperties.url, String identifier) {
+        return "${publicViewUrl}/${Base64.getEncoder().encodeToString(identifier.bytes)}" as String
+    }
+
+    void updateMultichoiceWithCurrentNode(MultichoiceMapField field, UriNode node) {
+        List<String> splitPathList = splitUriPath(node.uriPath)
+
+        change field value { splitPathList }
+        change field options { findOptionsBasedOnSelectedNode(node, splitPathList) }
+    }
+
+    List<String> splitUriPath(String uri) {
+        String rootUri = "/"
```

Comment:
I think that root URI should be always the same value as URI separator. It should be also changed in UriService. Both default root uri and name should be used from properties. Properties should have current values as default.